portalnet/src/overlay_service.rs

```diff
@@ -1671,9 +1706,23 @@ where
         let validated_content: Vec<(TContentKey, Vec<u8>)> = join_all(handles)
             .await
             .into_iter()
+            .enumerate()
             // Whether the spawn fails or the content fails validation, we don't want it:
-            .filter_map(|content| content.unwrap_or(None))
+            .filter_map(|(index, content)| content.unwrap_or_else(|err| {
+                // Extract the error from the thread handle
+                let err = err.into_panic();
+                let err = if let Some(err) = err.downcast_ref::<&'static str>() {
+                    err.to_string()
+                } else if let Some(err) = err.downcast_ref::<String>() {
+                    err.clone()
+                } else {
+                    format!("?{:?}", err)
```

Comment:
Is there an extra `?` in the message?

also... `format!("{err:?}")`

Reply:
Nice catch